src/main/java/org/elasticsearch/index/query/MatchAllQueryBuilder.java

```diff
@@ -100,15 +114,42 @@ public Query parse(QueryParseContext parseContext) throws IOException, QueryPars
                 }
             }
         }
+    }
 
-        if (boost == 1.0f) {
-            return Queries.newMatchAllQuery();
+    @Override
+    public void readFrom(StreamInput in) throws IOException {
+        this.boost = in.readFloat();
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeFloat(this.boost);
+    }
+    
+    @Override
+    public int hashCode() {
+        int hash = super.hashCode();
+        hash = maybeHashcode(hash, boost);
```

Comment:
seems odd that we have to call a method that accepts `Object` passing in a `float` as argument. Can we simplify?